consensus/types/src/block_contents.rs

```diff
@@ -0,0 +1,56 @@
+use crate::{
+    AbstractExecPayload, BeaconBlock, BeaconBlockAndBlobSidecars, BlobSidecars, EthSpec, ForkName,
+    ForkVersionDeserialize,
+};
+use derivative::Derivative;
+use serde_derive::{Deserialize, Serialize};
+
+/// A wrapper over a [`BeaconBlock`] or a [`BeaconBlockAndBlobSidecars`].
+#[derive(Clone, Debug, Derivative, Serialize, Deserialize)]
+#[derivative(PartialEq, Hash(bound = "T: EthSpec"))]
+#[serde(untagged)]
+#[serde(bound = "T: EthSpec")]
+pub enum BlockContents<T: EthSpec, Payload: AbstractExecPayload<T>> {
```

Comment:
This looks like a http api specific type. Should we have it in `eth2/types` instead of in `consensus/types`?

Reply:
Nice catch, thanks @pawanjay176 !